Sources/Extensions/UICollectionView+Registration.swift

```diff
@@ -37,6 +37,15 @@ public extension UICollectionView {
     }
 
     func registerView(by viewItem: CollectionViewReusableViewItem) {
```

Comment:
This function shouldn't take `CollectionViewReusableViewItem` as a parameter. Let's update it to something like this `func registerView(with type: ReuseType, kind: Kind)`. 

Also it would be great to rename function `func register(by type: ReuseType)` (line 28) to `func registerCell(by type: ReuseType)`. So both functions will have almost the same signature.

Reply:
Agree but think I can omit kind because `ReusableViewType` contain it.
`
func registerView(with type: ReusableViewType) {
 let kind = type.kind
....
}
`